lib/prom_ex/plugins/plug_cowboy.ex

```diff
@@ -0,0 +1,185 @@
+if Code.ensure_loaded?(Plug.Cowboy) do
+  defmodule PromEx.Plugins.PlugCowboy do
+    @moduledoc """
+    This plugin captures HTTP request metrics emitted by Plug.Cowboy.
+
+    This plugin exposes the following metric group:
+    - `:plug_cowboy_http_event_metrics`
+
+
+    To use plugin in your application, add the following to your PromEx module:
+    ```
+    defmodule WebApp.PromEx do
+      use PromEx, otp_app: :web_app
+
+      @impl true
+      def plugins do
+        [
+          ...
+          PromEx.Plugins.PlugCowboy
+        ]
+      end
+
+      @impl true
+      def dashboards do
+        [
+          ...
+          {:prom_ex, "plug_cowboy.json"}
+        ]
+      end
+    end
+    ```
+    """
+
+    use PromEx.Plugin
+    require Logger
+
+    @impl true
+    def event_metrics(opts) do
+      otp_app = Keyword.fetch!(opts, :otp_app)
+      metric_prefix = PromEx.metric_prefix(otp_app, :plug_cowboy)
+
+      [
+        http_events(metric_prefix, opts)
+      ]
+    end
+
+    defp http_events(metric_prefix, opts) do
+      # Shared configuration
+      cowboy_stop_event = [:cowboy, :request, :stop]
+      http_metrics_tags = [:status, :method, :path]
+      ignore_routes = Keyword.get(opts, :ignore_routes, [])
```

Comment:
Any reason that the ignored routes are in a keyword list? I imagine a list of regex/binaries would probably work better?

Reply:
makes more sense for it to be a simple list :)